crates/sui-json-rpc/src/error.rs

```diff
@@ -171,10 +171,17 @@ impl From<Error> for RpcError {
                         })
                         .collect();
 
-                    assert!(
-                        !new_errors.is_empty(),
-                        "NonRecoverableTransactionError should have at least one non-retryable error"
-                    );
+                    if new_errors.is_empty() {
+                        tracing::error!("NonRecoverableTransactionError should have at least one non-retryable error");
+                    }
+
+                    // TODO(wlmyng) verify that this is a valid condition to assert since we've
```

Comment:
I was informed that if we get this error, then there should be at least one error that is nonretryable. I use the SuiError.is_retryable method to determine recoverability, filtering out transient ones, but maybe that's not the correct method to use?

Reply:
@bmwill  do you have a link to the crashed node's log?

Reply:
```
2ae8a.., k#963c0871.., k#905d9f4b.., k#97d0d05b.., k#95aff462.., k#a3e5d733.., k#a6d0377c.., k#8c0dcf63.., k#8e7b6be6.., k#a4c899c1.., k#a5bd06fd.., k#b51b2f0b.., k#b0ce3137..]), (RpcError("transport error", "Unknown error"), 63, [k#abf207b5..])].
fullnode_1  | thread 'sui-node-runtime2023-08-31T12:34:24.605533Z ERROR json-rpc-request:execute_transaction_block{tx_bytes=Base64("AAARAQBjzuKkqx9FVzPC3umPSCL3XJlYzIjdfmjJquSXDOiUHvy0nAEAAAAAIFBU9+romMteeLNhykZRq0Vth5GTP3JMqW4YokHOTJ9XAQA8cmqauYLQ1zZv/8V131pL/7R1YCdiWjuqMuK4wrZbXfy0nAEAAAAAIB2IVUL3OIhVhpiNf0yVm/bo8wbjDy/TRKKtWef24CMdAAitvi0BAAAAAAEBLG/BK/DQk7U5HnwP7X4ETVK8FOsp9jUqP7NY4z6Acp716iYAAAAAAAEBAfV2KuWuGaIBa7Izxy2aSyy6WjAiN6gnJK9mKSrkOuUtj7wYAAAAAAABAARs7QAAAAEBAATQ4wAAAAEBAAgY74zKAQAAAAAIrb4tAQAAAAAACL18n7MBAAAAAAhYqB4BAAAAAAAgmiHVU472ZPTRK0SVR6yjZoKqsu5RBZ/Y5ofnVtRppXoACCjZlkuKAQAAAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgEAAAAAAAAAAAEB8c8OgQSN8Wjr6xuAMPrSSz4LU66CfCUFP/8HecFEW2+PvBgAAAAAAAAEAgABAQIABQACAQAAAQEABQABAwAAAAAA65IQ4pgEiRVMw8KTQyuaGxMA7dDVgP4iad2c2jS67m0QcG9zaXRpb25fbWFuYWdlcgRtaW50AwddH0fqabsN4xwxPXrPibiQ27iZHqjgPGw1UXH4S7G6SgZ0dXJib3MGVFVSQk9TAAcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgNzdWkDU1VJAAeRv7w4akGvz9myUzBY1+kVodOCkInMJo/0Mz1U1jOcoQpmZWUzMDAwYnBzCkZFRTMwMDBCUFMAEAEDAAEEAAIBAAICAAEFAAEGAAEHAAEIAAEJAAEKAAELAAEMAAENAAEOAAEPAAEQAJoh1VOO9mT00StElUeso2aCqrLuUQWf2OaH51bUaaV6AYWm8zLaRHbt/Wb9yIM+OmrdhlIMJiTAjNp56z3APo3Z/LScAQAAAAAgH1xnZo3iZz/Ymjcj2Mbpf3tVuBx3WF8h0V2WsPalePqaIdVTjvZk9NErRJVHrKNmgqqy7lEFn9jmh+dW1GmlevACAAAAAAAAcFQpAQAAAAAA") signatures=[Base64("AIQy4rt/EFnqdzmdGDaoCOvcmWHOfG6nb1sZvU/4fezVSMS3wHpQFuth4jGzi0An/fh6X3OxjP477/MUXuXQSQ/b/Y1LD1QWZdDhdQ+NSCzqm1/ye1rDSmHG7VRzmCA6KA==")] opts=Some(SuiTransactionBlockResponseOptions { show_input: true, show_raw_input: false, show_effects: true, show_events: true, show_object_changes: false, show_balance_changes: false }) request_type=Some(WaitForLocalExecution)}: telemetry_subscribers: panicked at 'NonRecoverableTransactionError should have at least one non-retryable error', crates/sui-json-rpc/src/error.rs:174:21 panic.file="crates/sui-json-rpc/src/error.rs" panic.line=174 panic.column=21
fullnode_1  | ' panicked at 'NonRecoverableTransactionError should have at least one non-retryable error', crates/sui-json-rpc/src/error.rs:174:21
fullnode_1  | note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
sui_fullnode_1 exited with code 139
```